lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/components/GraalVMComponentUpdater.scala

```diff
@@ -0,0 +1,101 @@
+package org.enso.runtimeversionmanager.components
+
+import java.nio.file.Path
+
+import org.enso.runtimeversionmanager.OS
+
+import scala.sys.process._
+import scala.util.{Success, Try}
+
+/** Module that manages components of the GraalVM distribution.
+  *
+  * @param runtime the GraalVM runtime
+  * @param os the operating system
+  */
+class GraalVMComponentUpdater(runtime: GraalRuntime, os: OS)
+    extends RuntimeComponentUpdater {
+
+  import GraalVMComponentUpdater._
+
+  /** List the installed GraalVM components.
+    *
+    * @return the list of installed GraalVM components
+    */
+  override def list: Try[Seq[GraalVMComponent]] = {
+    val suppressStderr = ProcessLogger(_ => ())
+    val process = Process(
+      Seq[String](Paths.gu, "list", "-v"),
+      Some(runtime.path.toFile),
+      ("JAVA_HOME", Paths.javaHome),
+      ("GRAALVM_HOME", Paths.javaHome)
+    )
+
+    for {
+      lines <- Try(process.lazyLines(suppressStderr))
+    } yield ListOut.parse(lines)
+  }
+
+  /** Install the provided GraalVM components.
+    *
+    * @param components the list of components to install
+    */
+  override def install(components: Seq[GraalVMComponent]): Try[Unit] = {
+    if (components.nonEmpty) {
+      val componentsList = components.map(_.id)
+      val process = Process(
+        Seq[String](Paths.gu, "install") ++ componentsList,
+        Some(runtime.path.toFile),
+        ("JAVA_HOME", Paths.javaHome),
+        ("GRAALVM_HOME", Paths.javaHome)
+      )
+      Try(process.!!)
+    } else {
+      Success(())
+    }
+  }
+
+  private object Paths {
+
+    /** Path to the java home directory in the runtime distribution. */
+    def javaHome: Path = os match {
+      case OS.Linux   => runtime.path
+      case OS.MacOS   => runtime.path / "Contents" / "Home"
+      case OS.Windows => runtime.path
+    }
```

Comment:
This is redundant to what is already in `GraalRuntime`.

Reply:
That explains why updater tests failed on macOS